patterns/structural/proxy.py

```diff
@@ -1,55 +1,67 @@
 """
+*What is this pattern about?
+Proxy is used in places where you want to add functionality to a class without
+changing its interface. The main class is called `Real Subject`. A client should
+use the proxy or the real subject without any code change, so both must have the
+same interface. Logging and controlling access to the real subject are some of
+the proxy pattern usages.
+
+*References:
+https://refactoring.guru/design-patterns/proxy/python/example
+https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Fronting.html
+
 *TL;DR
-Provides an interface to resource that is expensive to duplicate.
+Add functionality to a resource without changing its interface.
```

Comment:
I don't like how this phrase emphasises on adding functionality.
Although proxy can provide additional logic (caching, or checking permissions or other preconditions), but it is not important, and it is not visible for its client.